test/extended/cmd.sh

```diff
@@ -74,75 +75,74 @@ popd
 rm -rf $tmp
 
 # ensure a local-only image gets a docker image(not imagestream) reference created.
-[ "$(oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' ' | grep -E "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+")" ]
+current_project="$( oc project -q )"
+os::cmd::expect_success 'oc new-project test-scratchimage'
+os::cmd::expect_success 'oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker'
+os::cmd::expect_success_and_text 'oc describe bc' 'From Image:\s+DockerImage test/scratchimage:latest'
```

Comment:
I mean we were just approaching the problem incorrectly. We wanted to see if one field in one specific object was set to a specific value. In order to test this previously we were taking the entire YAML dump of every object that would have been created by `oc new-app` and doing the `tr` and `grep` thing. Instead, this is a more straight-forward test: we want to inspect the `BuildConfig`, so we do that.

I attempted with a Go template up to the point that I realized that we were relying on the ordering of items in the list in order to index into the correct one. The alternative was having logic on each list item and determining which type it was from the `ObjectMeta` ... I'm not certain this is a failure of the `jsonpath` or template system but instead just a poor approach in the first place.